packages/web-app-files/src/components/SideBar/Shares/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -1,32 +1,51 @@
 <template>
   <div>
-    <translate tag="label" for="files-collaborators-role-button" class="oc-label">Role</translate>
-    <oc-select
-      v-model="selectedRole"
-      input-id="files-collaborators-role-button"
-      class="files-collaborators-role-button-wrapper"
-      :options="roles"
-      :clearable="false"
-      label="label"
+    <hr />
+    <oc-button
+      id="files-collaborators-role-button"
+      appearance="raw"
+      justify-content="left"
+      gap-size="xsmall"
     >
-      <template #option="option">
-        <role-item :role="option" />
+      <translate v-if="isAdvancedRoleSelected" key="advanced-permissions-select"
+        >Invite with custom permissions</translate
+      >
+      <translate v-else key="role-select" :translate-params="{ name: selectedRole.inlineLabel }"
+        >Invite as %{ name }</translate
+      >
+      <oc-icon name="expand_more" />
+    </oc-button>
+    <oc-drop ref="rolesDrop" toggle="#files-collaborators-role-button" mode="click" close-on-click>
+      <template #special>
+        <oc-list class="files-recipient-role-drop-list" :aria-label="rolesListAriaLabel">
+          <li v-for="role in roles" :key="role.name">
+            <oc-button
```

Comment:
Not sure how accessible this solution is as of now 🤔 let's talk about this in the web call today